pkg/cmd/cli/cli.go

```diff
@@ -133,28 +135,60 @@ func MapPrint(content map[string]interface{}) {
 		}
 		fmt.Printf("%s\n", string(JSON))
 	} else {
-		keys := make([]string, 0, len(content))
-		for k := range content {
-			keys = append(keys, k)
+		hosts := make([]string, 0, len(content))
+		for host := range content {
+			hosts = append(hosts, host)
 		}
-		sort.Strings(keys)
-		for _, k := range keys {
-			fmt.Printf("%s:\n", k)
-
-			// Warning, the struct fields must be exported!
-			s := content[k]
-			v := reflect.ValueOf(s)
-			typeOfS := v.Type()
-			for i := 0; i < v.NumField(); i++ {
-				if v.Field(i).Interface() != nil {
-					fmt.Printf(
-						"\t%-20s: %-60s\n",
-						typeOfS.Field(i).Name,
-						v.Field(i).Interface(),
-					)
+		sort.Strings(hosts)
+		for _, h := range hosts {
+			if content[h] != nil {
+				v := reflect.ValueOf(content[h])
+				typeOfS := v.Type()
+				// print the host
+				fmt.Printf("%s:\n", h)
+				// print the content dependent upon its type
+				switch t := content[h].(type) {
+				case redfish.SettingsAttributes:
+					for setting, value := range t {
+						fmt.Printf("\t%-40s: %+v\n", setting, value)
+					}
+				case State:
+					for i := 0; i < v.NumField(); i++ {
+						if v.Field(i).Interface() != nil {
+							fmt.Printf(
+								"\t%-20s: %-60s\n",
+								typeOfS.Field(i).Name,
+								v.Field(i).Interface(),
+							)
+						}
+					}
+				case StateChange:
+					for i := 0; i < v.NumField(); i++ {
+						if v.Field(i).Interface() != nil {
+							fmt.Printf(
+								"\t%-20s: %-60s\n",
+								typeOfS.Field(i).Name,
+								v.Field(i).Interface(),
+							)
+						}
+					}
```

Comment:
```suggestion
				case StateChange, Boot, Override:
 					for i := 0; i < v.NumField(); i++ {
 						if v.Field(i).Interface() != nil {
 							fmt.Printf(
 								"\t%-20s: %-v\n",
 								typeOfS.Field(i).Name,
 								v.Field(i).Interface(),
 							)
 						}
 					}
```

Reply:
Looks like I accepted this prematurely, it seems to have broken the build.
```
pkg/cmd/cli/cli.go:165:23: undefined: Boot
pkg/cmd/cli/cli.go:165:29: undefined: Override
```

Reply:
Upon fixing this I stumbled onto the `MapPrint` changes.
I'm giving polymorphism a try, before giving up. There isn't a huge rush to get this in, so I can spend a little time on it.